sources/api/schnauzer/src/helpers/mod.rs

```diff
@@ -2058,6 +2078,21 @@ mod test_oci_spec {
         }
     }
 
+    #[test]
+    fn oci_spec_resource_limits_adjust_for_efa_test() {
+        let json = json!({"max-open-files": {"hard-limit": 1, "soft-limit": 2}});
+        let mut rlimits = oci_spec_resource_limits(&json).unwrap();
+        adjust_resource_limits_for_efa(&mut rlimits);
+        let rendered = Containerd::get_resource_limits(
+            &MaxLockedMemory,
+            rlimits.get(&MaxLockedMemory).unwrap(),
+        );
```

Comment:
Can you add a test for the other case, where there's no EFA device?

Reply:
Sure thing.

Reply:
I had to implement a trait that wraps the `pciclient::is_efa_attached()` call so that I can unit test this for both cases. 

Let me know if you like it or not.